src/main/java/liquibase/ext/opensearch/changelog/OpenSearchHistoryService.java

```diff
@@ -0,0 +1,223 @@
+        }
+
+        try {
+            this.getOpenSearchClient()
+                    .create(r -> r.index(this.getDatabaseChangeLogTableName())
+                            .id(ranChangeSet.getId())
+                            .document(ranChangeSet));
+        } catch (final IOException e) {
+            throw new DatabaseException(e);
+        }
+    }
+
+    @Override
+    protected void removeRanChangeSet(final ChangeSet changeSet) throws DatabaseException {
+        try {
+            this.getOpenSearchClient()
+                    .delete(r -> r.index(this.getDatabaseChangeLogTableName()).id(String.valueOf(changeSet.getId())));
+        } catch (final IOException e) {
+            throw new DatabaseException(e);
+        }
+    }
+
+    @Override
+    public void clearAllCheckSums() throws DatabaseException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected long countTags(final String tag) throws DatabaseException {
+        final var request = new SearchRequest.Builder()
+                .index(this.getDatabaseChangeLogTableName())
+                .query(q -> q.match(m -> m.field("tag").query(FieldValue.of(tag))))
+                .build();
+        try {
+            final var response = this.getOpenSearchClient().search(request, RanChangeSet.class);
+            return response.hits().total().value();
+        } catch (final IOException e) {
+            throw new DatabaseException(e);
+        }
+    }
+
+    @Override
+    protected void tagLast(final String tagString) throws DatabaseException {
+        // TODO
+    }
+
+    @Override
+    protected long countRanChangeSets() throws DatabaseException {
+        return this.queryRanChangeSets().size();
+    }
+
+    @Override
+    protected void updateCheckSum(final ChangeSet changeSet) throws DatabaseException {
+        @AllArgsConstructor
+        class CheckSumObj {
+            final CheckSum lastCheckSum;
+        }
+        final var currentChecksumVersion = Optional.ofNullable(changeSet.getStoredCheckSum())
+                .map(cs -> ChecksumVersion.enumFromChecksumVersion(cs.getVersion()))
+                .orElse(ChecksumVersion.latest());
+        final var checkSum = changeSet.generateCheckSum(currentChecksumVersion);
+
+        try {
+            this.getOpenSearchClient()
+                    .update(r -> r.index(this.getDatabaseChangeLogTableName())
+                                    .id(changeSet.getId())
+                                    .doc(new CheckSumObj(checkSum))
+                            , RanChangeSet.class);
+        } catch (final IOException e) {
+            throw new DatabaseException(e);
+        }
+    }
+
+    @Override
+    public boolean supports(final Database database) {
+        return OpenSearchLiquibaseDatabase.PRODUCT_NAME.equals(database.getDatabaseProductName());
+    }
+
+    @Override
+    public boolean isDatabaseChecksumsCompatible() {
+        return false;
```

Comment:
Should be true. It is used when we upgrade checksums versions . In core we check if any checksum is using a previous checksum version, if so it's false. But by now it can be just true.